tests/pebble.rs

```diff
@@ -783,6 +805,7 @@ struct EnvironmentConfig {
     ///
     /// See <https://github.com/letsencrypt/pebble?tab=readme-ov-file#valid-authorization-reuse>
     authz_reuse: u8,
+    retry_policy: RetryPolicy,
```

Comment:
If we're going to do this, it should be in a separate commit, and we should just delete the `RETRY_POLICY` `const` in favor of the `EnvironmentConfig::default()` impl.